### DISCUSSION TITLE
Intended popup behaviour with edit-text widget, with cancelPopups="no"?

### DISCUSSION THREAD

**Discussion by clutterstack** [TiddlyWiki/TiddlyWiki5#5951] [Q&A]
I'm playing with popups, and I see that if I invoke a popup with a button, then while that one is open, try to invoke one by focusing an edit-text widget, the edit-text widget's popup appears momentarily, then vanishes (and the pre-existing popup remains).

Setting the cancelPopups attribute of the edit-text widget to "yes" prevents this, but I'm wondering if there's an intended use-case for the default scenario.

**Comment by Jermolene** [comment]
Hi @clutterstack how are you invoking the popup? Using the `popup=foo` attribute? Do you get the same behaviour on v5.2.0-prerelease and the current v5.1.23 version?

**Reply by clutterstack** [reply]
Hi Jeremy,

I was using `popup=foo` with the button widget, but it works the same if I get the sidebar search to show a dropdown. So I paste the following in a new tiddler either at https://tiddlywiki.com/ or at https://tiddlywiki.com/prerelease/, put a few characters into the search to get a results popup, and click into the edit-text widget's input element.

I didn't catch this before, but the behaviour of the edit-text popup with a preexisting popup is very like its general behaviour if  `class="tc-popup-handle"` is omitted: the popup disappears on release of the mouse button. 

```
<$vars popstate=<<qualify "testpopstate">>>
<$edit-text tiddler=<<qualify "$:/state/test/edittext-text">> class="tc-popup-handle" focusPopup=<<popstate>> tag="input"/>

<$reveal state=<<popstate>> type="popup">Edit-text's popup
</$reveal>

<<popstate>>: {{{[<popstate>get[text]]}}}<br>

</$vars>
```

I was wondering whether it has something to do with the click handler that's added at the end of `Popup.prototype.show()`. If I'm reading this right:

If there's an existing popup before the edit-text one is added on focus, then if `
Popup.prototype.handleEvent()` gets invoked somehow on mouseup, we could have `cancel(level)` invoked when `level=1` and `numPopups=2`, which then would pop the last popup  -- the edit-text one -- off the popups array and delete its tiddler.

If we had no pre-existing popups and omit the `class="tc-popup-handle"`, then we'd have `level=0` and `numPopups=1`, which results in the same thing if `cancel()` gets invoked on mouseup: losing the popup. 

I didn't think the click event would fire if the handler was added after the mousedown -- but I'm not confident, and  haven't gone through all the code for the editor.

Edit: grammar!

**Reply by clutterstack** [reply]
@Jermolene I should point out that I don't need popups from the edit-text widget without `cancelPopups="yes"`. I was just trying to understand popups, and this seemed odd.
